packages/arb-token-bridge-ui/src/pages/api/teleports/eth.ts

```diff
@@ -0,0 +1,157 @@
+import { gql } from '@apollo/client'
+import {
+  getL1SubgraphClient,
+  getSourceFromSubgraphClient
+} from '../../../api-utils/ServerSubgraphUtils'
+import { getInboxAddressFromOrbitChainId } from '../../../util/orbitChainsList'
+import { FetchEthTeleportsFromSubgraphResult } from '../../../util/teleports/fetchEthTeleportsFromSubgraph'
+import { ChainId } from '../../../util/networks'
+
+// Extending the standard NextJs request with Deposit-params
+type NextApiRequestWithDepositParams = NextApiRequest & {
+  query: {
+    sender?: string
+    receiver?: string
+    l1ChainId: string
+    l2ChainId: string
+    l3ChainId: string
+    search?: string
+    page?: string
+    pageSize?: string
+    fromBlock?: string
+    toBlock?: string
+  }
+}
+
+type EthTeleportResponse = {
+  meta?: { source: string | null }
+  data: FetchEthTeleportsFromSubgraphResult[]
+  message?: string // in case of any error
+}
+
+export default async function handler(
+  req: NextApiRequestWithDepositParams,
+  res: NextApiResponse<EthTeleportResponse>
+) {
+  try {
+    const {
+      sender,
+      receiver,
+      search = '',
+      l1ChainId,
+      l2ChainId,
+      l3ChainId,
+      page = '0',
+      pageSize = '10',
+      fromBlock,
+      toBlock
+    } = req.query
+
+    // validate method
+    if (req.method !== 'GET') {
+      res
+        .status(400)
+        .send({ message: `invalid_method: ${req.method}`, data: [] })
+      return
+    }
+
+    // validate the request parameters
+    const errorMessage = []
+
+    if (!l1ChainId) errorMessage.push('<l1ChainId> is required')
+
+    if (!l2ChainId) errorMessage.push('<l2ChainId> is required')
+
+    if (!l3ChainId) errorMessage.push('<l3ChainId> is required')
+
+    if (!sender && !receiver)
+      errorMessage.push('<sender> or <receiver> is required')
+
+    if (errorMessage.length) {
+      res.status(400).json({
+        message: `incomplete request: ${errorMessage.join(', ')}`,
+        data: []
+      })
+      return
+    }
+
+    // if invalid pageSize, send empty data instead of error
+    if (isNaN(Number(pageSize)) || Number(pageSize) === 0) {
+      res.status(200).json({
+        data: []
+      })
+      return
+    }
+
+    let subgraphClient
+    try {
+      subgraphClient = getL1SubgraphClient(Number(l2ChainId))
+    } catch (error: any) {
+      // catch attempt to query unsupported networks and throw a 400
+      res.status(400).json({
+        message: error?.message ?? 'Something went wrong',
+        data: []
+      })
+      return
+    }
+
+    const l3InboxAddress = getInboxAddressFromOrbitChainId(Number(l3ChainId))
+
+    if (typeof l3InboxAddress === 'undefined') {
+      res.status(400).json({
+        message: `inbox address not found for chain-id: ${l1ChainId} -> ${l2ChainId} -> ${l3ChainId}`,
+        data: []
+      })
+      return
+    }
+
+    const createRetryableFunctionSelector = '0x679b6ded'
+
+    const query = `{
```

Comment:
Fetching ETH teleports is not straight forward. We query the L2 retryables:
1. created by the sender
2. where the calldata contains `submitRetryable` function
3. and the destination address is the inbox address of our teleport-enabled chain.

Using this, we already know the `l3chainId` of the teleport.
However, we get very basic information from subgraphs - enough to dedupe and sort. Other information required by the tx to be displayed on UI needs to be extracted per txn, which is done in `transformTransaction` function.